Gifski/ExportModifiedVideo.swift

```diff
@@ -0,0 +1,323 @@
+			switch state {
+			case let .empty(error: error):
+				if let error {
+					Text(error)
+				}
+				Button("Cancel") {
+					dismiss()
+				}
+			case .generating, .finished:
+				ProgressView()
+				if input.gifDuration >= 20 {
+					Button("Cancel") {
+						dismiss()
+					}
+				}
+			}
+		}
+		.padding()
+		.task(priority: .medium) {
+			do {
+				let task = Self.exportModifiedVideo(input: input)
+				self.state = .generating(progress: 0)
+				await withTaskCancellationHandler {
+					for await progress in task.progress {
+						self.state = .generating(progress: progress)
+					}
+				} onCancel: {
+					task.cancel()
+				}
+				try Task.checkCancellation()
+				let (outputURL, aspectRatio) = try await task.value
+				try Task.checkCancellation()
+
+				// let the animation play
+				self.state = .generating(progress: 1)
+				try await Task.sleep(for: .seconds(1.0))
+
+				self.state = .finished(
+					filename: input.conversion.sourceURL.filenameWithoutExtension + " modified.mp4",
+					url: outputURL,
+					player: AVPlayer(url: outputURL),
+					aspectRatio: aspectRatio
+				)
+			} catch {
+				if Task.isCancelled || error.isCancelled {
+					return
+				}
+				self.state = .empty(error: error.localizedDescription)
+			}
+		}
+		.fileExporter(
+			isPresented: .init(get: {
+				if case .finished = state {
+					return true
+				}
+				return false
+			}, set: { _ in }),
+			item: exportableMP4,
+			defaultFilename: defaultFileName
+		) {
+			do {
+				let url = try $0.get()
+				try? url.setAppAsItemCreator()
+				dismiss()
+			} catch {
+				state = .empty(error: error.localizedDescription)
+			}
+		} onCancellation: {
+			dismiss()
+		}
+		.fileDialogCustomizationID("export")
+		.fileDialogMessage("Choose where to save the video")
+		.fileDialogConfirmationLabel("Save")
+		.onDisappear {
+			removeTempExport()
+		}
+	}
+	struct Input {
+		let conversion: GIFGenerator.Conversion
+		let audioAssets: [AVAsset]
+		let gifDuration: Double
+
+		init(conversion: GIFGenerator.Conversion, audioAssets: [AVAsset], speed: Double, assetDuration: TimeInterval) {
+			self.conversion = conversion
+			self.audioAssets = audioAssets
+			let range = self.conversion.timeRange ?? 0...assetDuration
+			gifDuration = range.length * speed
+		}
+	}
+
+	private var defaultFileName: String {
+		switch state {
+		case let .finished(filename, _, _, _):
+			filename
+		default:
+			"Untitled.mp4"
+		}
+	}
+
+	private func removeTempExport() {
+		exportURL.map {
+			try? FileManager.default.removeItem(at: $0)
+		}
+	}
+
+	private var exportableMP4: ExportableMP4? {
+		exportURL.map {
+			ExportableMP4(url: $0)
+		}
+	}
+
+	private var exportURL: URL? {
+		switch state {
+		case let .finished(_, url, _, _):
+			url
+		default:
+			nil
+		}
+	}
+
+	private enum ConvertState: Equatable, Sendable {
+		case empty(error: String?)
+		case generating(progress: Double)
+		case finished(filename: String, url: URL, player: AVPlayer, aspectRatio: Double)
```

Comment:
```suggestion
		case finished(filename: String, url: URL, aspectRatio: Double)
```

Make sure there are no leftovers.